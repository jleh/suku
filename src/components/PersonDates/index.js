import React from 'react';

import { renderAge } from '../../util';

const renderPlace = (place) => (place ? place.name : '');

const PersonDates = ({ events }) => {
  const { birth, death, birthPlace, deathPlace, birthISO, deathISO } = events;

  return (
    <div>
      {birth && (
        <div>
          * {birth} {renderPlace(birthPlace)}
        </div>
      )}
      {death && (
        <div>
          â€  {death} {renderPlace(deathPlace)} {renderAge(deathISO, birthISO)}
        </div>
      )}
    </div>
  );
};

export default PersonDates;
