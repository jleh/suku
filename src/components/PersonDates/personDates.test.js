import React from 'react';
import renderer from 'react-test-renderer';

import PersonDates from '.';

const renderComponent = events => renderer
  .create(<PersonDates events={events} />)
  .toJSON();

test('Renders person birth and death', () => {
  const events = {
    birth: '6.6.1800',
    death: '9.9.1890',
    birthPlace: { name: 'Place 1' },
    deathPlace: { name: 'Place 2' }
  };

  expect(renderComponent(events)).toMatchSnapshot();
});

test('Renders living person', () => {
  const events = {
    birth: '6.6.1800',
    birthPlace: { name: 'Place 1' }
  };

  expect(renderComponent(events)).toMatchSnapshot();
});

test('Renders person whose death date is only known', () => {
  const events = {
    death: '9.9.1890'
  };

  expect(renderComponent(events)).toMatchSnapshot();
});
