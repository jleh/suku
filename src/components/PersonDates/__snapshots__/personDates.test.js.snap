// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders living person 1`] = `
<div>
  <div>
    * 
    6.6.1800
     
    Place 1
  </div>
</div>
`;

exports[`Renders person birth and death 1`] = `
<div>
  <div>
    * 
    6.6.1800
     
    Place 1
  </div>
  <div>
    † 
    9.9.1890
     
    Place 2
     
    90 vuotta
  </div>
</div>
`;

exports[`Renders person whose death date is only known 1`] = `
<div>
  <div>
    † 
    9.9.1890
     
    
     
  </div>
</div>
`;
